packages/main/src/Calendar.js

```diff
@@ -416,11 +468,83 @@ class Calendar extends UI5Element {
 				this._hideMonthPicker();
 			}
 		}
+
+		if (isTabNext(event)) {
+			const header = this.shadowRoot.querySelector("ui5-calendar-header");
+			const monthButton = header.shadowRoot.querySelector("[data-sap-show-picker='Month']");
+			const yearButton = 	header.shadowRoot.querySelector("[data-sap-show-picker='Year']");
+			const target = event.target;
+
+			if (target.tagName === "UI5-DAYPICKER" || target.tagName === "UI5-MONTHPICKER" || target.tagName === "UI5-YEARPICKER") {
+				if (monthButton.getAttribute("hidden") === null) {
+					monthButton.focus();
+				} else {
+					yearButton.focus();
+				}
+				event.preventDefault();
+			} else if (event.target.tagName === "UI5-CALENDAR-HEADER" && event.path[0].getAttribute("data-sap-show-picker") === "Month") {
+				yearButton.focus();
+				event.preventDefault();
+			} else {
+				this._setPickerCurrentTabindex(-1);
+			}
+		}
+
+		if (isTabPrevious(event)) {
+			const header = this.shadowRoot.querySelector("ui5-calendar-header");
+			if (event.target.tagName === "UI5-CALENDAR-HEADER" && event.path[0].getAttribute("data-sap-show-picker") === "Month") {
+				this._moveFocusToPickerContent();
+				event.preventDefault();
+			} else if (event.target.tagName === "UI5-CALENDAR-HEADER" && event.path[0].getAttribute("data-sap-show-picker") === "Year") {
+				const monthButton = header.shadowRoot.querySelector("[data-sap-show-picker='Month']");
+				if (monthButton.getAttribute("hidden") === null) {
+					monthButton.focus();
+				} else {
+					this._moveFocusToPickerContent();
+				}
+				event.preventDefault();
+			}
+		}
+	}
+
+	_moveFocusToPickerContent() {
+		if (!this._oMonth._hidden) {
+			this.shadowRoot.querySelector("ui5-daypicker")._itemNav.focusCurrent();
```

Comment:
create getters for those:
 this.shadowRoot.querySelector("ui5-daypicker")
 this.shadowRoot.querySelector("ui5-monthpicker")
 this.shadowRoot.querySelector("ui5-yearpicker")

get dayPicker() {
return  this.shadowRoot.querySelector("ui5-daypicker")
}

Then use everywhere
this.dayPicker

Reply:
Done